uniswap-v2/src/modules/7_store_usd_prices.rs

```diff
@@ -0,0 +1,170 @@
+                let ordinal = event.log_ordinal;
+
+                let pool =
+                    pool_store.must_get_last(StoreKey::Pool.get_unique_pool_key(&event.pool));
+
+                let token0 = pool.token0_ref();
+                let token1 = pool.token1_ref();
+
+                let reserve0 = BigInt::from_str(sync.reserve0.as_str())
+                    .unwrap()
+                    .to_decimal(token0.decimals);
+                let reserve1 = BigInt::from_str(sync.reserve1.as_str())
+                    .unwrap()
+                    .to_decimal(token1.decimals);
+
+                if reserve0.is_zero() || reserve1.is_zero() {
+                    continue;
+                }
+
+                let token0_derived_price = reserve1.clone().div(reserve0.clone());
+                let token1_derived_price = reserve0.clone().div(reserve1.clone());
+
+                if utils::is_pricing_asset(&token1.address) {
+                    let mut token0_price = BigDecimal::zero();
+
+                    let token1_price =
+                        get_price_from_native_store(ordinal, &token1.address, &native_prices_store);
+
+                    if reserve1
+                        .mul(token1_price.clone())
+                        .ge(&min_liquidity_threshold)
+                    {
+                        token0_price = token0_derived_price * token1_price;
+                    }
+
+                    output_store.set(
+                        ordinal,
+                        StoreKey::TokenPrice.get_unique_pool_key(&token0.address),
+                        &token0_price,
+                    );
+                }
+
+                if utils::is_pricing_asset(&token0.address) {
+                    let mut token1_price = BigDecimal::zero();
+
+                    let token0_price =
+                        get_price_from_native_store(ordinal, &token0.address, &native_prices_store);
+
+                    if reserve0
+                        .mul(token0_price.clone())
+                        .ge(&min_liquidity_threshold)
+                    {
+                        token1_price = token1_derived_price * token0_price
+                    }
+
+                    output_store.set(
+                        ordinal,
+                        StoreKey::TokenPrice.get_unique_pool_key(&token1.address),
+                        &token1_price,
+                    );
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+fn get_price_from_native_store(
+    ordinal: u64,
+    token_address: &String,
+    native_prices_store: &StoreGetBigDecimal,
+) -> BigDecimal {
+    if token_address.eq(constants::WETH_ADDRESS) {
+        return get_weth_price_in_usd(ordinal, native_prices_store);
+    }
+    if constants::STABLE_COINS.contains(&token_address.as_str()) {
+        return BigDecimal::one();
+    }
+
+    for address in constants::PAIR_COINS.into_iter() {
+        let whitelisted_token = address.to_string();
```

Comment:
whitelisted_token is only used by reference from here. Could you just use `let whitelisted_token = address;` instead?